File: backend/app/schemas/goal.py
"""Goal schemas."""

import uuid
from datetime import date, time, datetime
from pydantic import BaseModel


class GoalCreate(BaseModel):
    title: str
    subject: str
    date: date
    semester_goal_id: uuid.UUID | None = None
    focus_window_start: time | None = None
    focus_window_end: time | None = None


class GoalResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    subject: str
    date: date
    is_completed: bool
    semester_goal_id: uuid.UUID | None
    focus_window_start: time | None
    focus_window_end: time | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/lib/models/goal.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'goal.freezed.dart';
part 'goal.g.dart';

@freezed
class Goal with _$Goal {
  const Goal._();
  const factory Goal({    required String id,
    required String userId,
    required String title,
    required String subject,
    required String date,
    @Default(false) bool isCompleted,
    String? semesterGoalId,
    String? focusWindowStart,
    String? focusWindowEnd,
    required DateTime createdAt,
    DateTime? syncedAt,
  }) = _Goal;

  factory Goal.fromJson(Map<String, dynamic> json) => _$GoalFromJson(json);
}
